### PR TITLE
feat: Add blog post for Node.js builder

### PR DESCRIPTION
**Pull request by ianlewis**
Adds a blog post about npm provenance beta and announcing beta for the SLSA3 Node.js builder.

### CODE REVIEW

File: docs/_posts/2023-05-01-bringing-improved-supply-chain-security-to-the-nodejs-ecosystem.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+untrusted and potentially compromised code from dependencies and other sources.
+While it does require a specific attack, this in turn could allow attackers to
+leak private keys or modify the provenance. As we have seen
+[in](https://github.com/advisories/GHSA-pjwm-rvh2-c87w)
+[the](https://github.com/advisories/GHSA-73qr-pfmq-6rp8)
+[past](https://github.com/advisories/GHSA-g2q5-5433-rhrf), open source packages
+are not immune to these targeted attacks.
+
+The Node.js builder, built in collaboration with GitHub, does this by executing
+the build and generation and signing of provenance in separate job steps
```
Comment (mlieberman85) [inline docs/_posts/2023-05-01-bringing-improved-supply-chain-security-to-the-nodejs-ecosystem.md:72]: This sentence is a bit unclear. I think it should just be "the build, generation, and signing of provenance in separate job steps"

**Review by MarkLodato — COMMENTED** [review]
Overall LGTM, just an unfinished sentence and a few nits.

File: docs/_posts/2023-05-01-bringing-improved-supply-chain-security-to-the-nodejs-ecosystem.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+First, it provides a way for package maintainers to generate provenance for
+their packages with minimal changes to their development workflows. This allows
+busy Open Source developers to get large supply chain security gains with only a
+small time investment.
+
+Second, it provides a natural way to host provenance along with the package
+itself. This takes advantage of existing package tooling to transparently
+distribute provenance to users so it can be verified easily.
+
+Third, Verification can be easily incorporated into the user's normal workflow
```
Comment (pnacht) [inline docs/_posts/2023-05-01-bringing-improved-supply-chain-security-to-the-nodejs-ecosystem.md:44]:
```suggestion
Third, verification can be easily incorporated into the user's normal workflow
```

File: docs/_posts/2023-05-01-bringing-improved-supply-chain-security-to-the-nodejs-ecosystem.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+via the `npm audit signatures` command. Provenance adds the most value if it can
+be verified before install. Support for verification in the `npm` CLI makes this
```
Comment (pnacht) [inline docs/_posts/2023-05-01-bringing-improved-supply-chain-security-to-the-nodejs-ecosystem.md:46]: Here we have "`npm` CLI", but the first paragraph (currently L9-10) has "npm CLI".

**Comment by ianlewis** [comment]
Still have some release related things to work out before this post will be ready. Reverting to draft to reflect that.

File: docs/_posts/2023-05-01-bringing-improved-supply-chain-security-to-the-nodejs-ecosystem.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+With this new support the provenance generated by npm projects achieves
+[SLSA Build Level 2](https://slsa.dev/spec/v1.0/levels#build-l2). SLSA Build L2
+gives projects a huge boost to software supply chain integrity of npm packages.
+It prevents tampering after the build with digital signatures, reduces the
+attack surface to one that can be more easily audited and hardened, and allows
+for easier migration to more hardened SLSA Build levels in the future.
+
+## Going Farther with SLSA on GitHub Actions
+
+[SLSA Build L3](https://slsa.dev/spec/v1.0/levels#build-l3-hardened-builds)
```
[the remark this answers is not in the document]
Comment (ianlewis) [reply docs/_posts/2023-05-01-bringing-improved-supply-chain-security-to-the-nodejs-ecosystem.md]: Done.

File: docs/_posts/2023-05-01-bringing-improved-supply-chain-security-to-the-nodejs-ecosystem.md (hunk 1 above)
Comment (ianlewis) [reply docs/_posts/2023-05-01-bringing-improved-supply-chain-security-to-the-nodejs-ecosystem.md]: Updated.

File: docs/_posts/2023-05-01-bringing-improved-supply-chain-security-to-the-nodejs-ecosystem.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ianlewis) [reply docs/_posts/2023-05-01-bringing-improved-supply-chain-security-to-the-nodejs-ecosystem.md]: Updated.

File: docs/_posts/2023-05-01-bringing-improved-supply-chain-security-to-the-nodejs-ecosystem.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+Second, it provides a natural way to host provenance along with the package
+itself. This takes advantage of existing package tooling to transparently
+distribute provenance to users so it can be verified easily.
+
+Third, Verification can be easily incorporated into the user's normal workflow
+via the `npm audit signatures` command. Provenance adds the most value if it can
+be verified before install. Support for verification in the `npm` CLI makes this
+
+With this new support the provenance generated by npm projects achieves
+[SLSA Build Level 2](https://slsa.dev/spec/v1.0/levels#build-l2). SLSA Build L2
```
[the remark this answers is not in the document]
Comment (ianlewis) [reply docs/_posts/2023-05-01-bringing-improved-supply-chain-security-to-the-nodejs-ecosystem.md]: Done.

File: docs/_posts/2023-05-01-bringing-improved-supply-chain-security-to-the-nodejs-ecosystem.md (hunk 2 above)
Comment (ianlewis) [reply docs/_posts/2023-05-01-bringing-improved-supply-chain-security-to-the-nodejs-ecosystem.md]: Done.

File: docs/_posts/2023-05-01-bringing-improved-supply-chain-security-to-the-nodejs-ecosystem.md (hunk 3 above)
Comment (ianlewis) [reply docs/_posts/2023-05-01-bringing-improved-supply-chain-security-to-the-nodejs-ecosystem.md]: Done.

**Review by ianlewis — COMMENTED** [review]
Made quite a few updates. Please take another look.

**Review by joshuagl — APPROVED** [review]
Very nice release! 🎉

**Review by mlieberman85 — APPROVED** [review]
Good stuff. Looking forward to this.

**Review by pnacht — APPROVED** [review]

**Comment by ianlewis** [comment]
A few things came up during a design review and we decided to address them before release. So we'll hold off on merging this blog post for now.

**Comment by ianlewis** [comment]
I think we'd like to target May 8 or 9 during OSS@NA as the publish date.

**Pull request merged**